prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1612,6 +1613,23 @@ void rxr_ep_progress_internal(struct rxr_ep *ep)
 
 	rxr_ep_check_peer_backoff_timer(ep);
 
+	/*
+	 * Send any queued handshake packet.
```

Comment:
This doesn't match with the list you're iterating as the peers are queued not the packets. Perhaps change to

"Resend handshake packet for any peers where the first handshake send failed."